internal/component/pyroscope/java/loop.go

```diff
@@ -63,6 +71,16 @@ func newProfilingLoop(pid int, target discovery.Target, logger log.Logger, profi
 		return p
 	}
 
+	p.wg.Add(1)
+	go func() {
```

Comment:
Should this happen in a goroutine? I would expect that we want cleanup to _complete_ before the regular loop begins, so starting this and allowing the internal scheduler to decide what goes first doesn't seem right.

Reply:
Hi @dehaansa, I was thinking the clean-up can safely run in parallel to the loop as a small optimization. It has a check to make sure the JFR file eventually created by the loop isn't accidentally deleted during clean-up, since that file always has the same name for the current process running alloy. 

Let me know if you think it's still worth changing though!